File: Java/NetBeans/QLTV/src/qltv/TaiLieu.java
package qltv;

public class TaiLieu {
    protected String maTL;
    protected String tenTL;
    protected int viTri;
    
    public TaiLieu(String maTL, String tenTL, int viTri) {
        this.maTL = maTL;
        this.tenTL = tenTL;
        this.viTri = viTri;
    }
    
    public void hienThi() {
        System.out.println("Ma TL: " + maTL);
        System.out.println("Ten TL: " + tenTL);
        System.out.println("Vi tri: " + viTri);
    }  
}

File: C#/QLThuVien/QLThuVien/TaiLieu.cs
namespace QLThuVien
{
    abstract class TaiLieu
    {
        protected string maTL;
        protected string tenTL;
        protected int viTri;

        public string MaTL { get => maTL; set => maTL = value; }
        public string TenTL { get => tenTL; set => tenTL = value; }

        public TaiLieu(string maTL, string tenTL, int viTri)
        {
            this.maTL = maTL;
            this.tenTL = tenTL;
            this.viTri = viTri;
        }

        public virtual void hienThi()
        {
            Console.WriteLine($"Ma tai lieu: {maTL}");
            Console.WriteLine($"Ten tai lieu: {tenTL}");
            Console.WriteLine($"Vi tri: {viTri}");
        }

        abstract public string loaiTL();
    }
}